File: iPhone/VerBank-CSTSv3-IOSClientApi/CSTS3Comm/ipop/OPFather.m
//
//  OPFather.m
//  JEDIv7-CSTSv3-ClientAPI
//
//  Created by baolin on 13/5/6.
//  Copyright (c) 2013年 wangyubo. All rights reserved.
//

#import "OPFather.h"
#import "JEDISystem.h"
#import "JEDIDateTime.h"

//static NSString * gs_opFather_jsonId    = @"opf";
static NSString * gs_opFather_id        = @"-1";
static NSString * gs_opFather_operateId = @"-2";
static NSString * gs_opFather_succeed   = @"-3";
static NSString * gs_opFather_errCode   = @"-4";
static NSString * gs_opFather_errMessage= @"-5";
static NSString * gs_opFather_usedTime  = @"-6";

@implementation OPFather

//------------------------------------------------------------------
-(id) init
{
    self = [super init];
    return self;
}

//------------------------------------------------------------------
-(id) initWithIp:(IPFather *)ip
{
    self = [super init];
    
    if (self != nil)
    {
        [super setEntry:gs_opFather_id entry:[ip getID]];
        [super setEntry:gs_opFather_operateId entry:[ip getOperateId]];
    }
    
    return self;
}

//------------------------------------------------------------------
-(NSString *) getID
{
    JEDI_SYS_Try
    {
        return [super getEntryString:gs_opFather_id];
    }
    JEDI_SYS_EndTry
 
    return nil;
}

//------------------------------------------------------------------
-(NSString *) getOperateId
{
    JEDI_SYS_Try
    {
        return [super getEntryString:gs_opFather_operateId];
    }
    JEDI_SYS_EndTry
    
    return nil;
}

//------------------------------------------------------------------
-(NSString *) getErrCode
{
    JEDI_SYS_Try
    {
        return [super getEntryString:gs_opFather_errCode];
    }
    JEDI_SYS_EndTry
    
    return nil;
}

//------------------------------------------------------------------
-(void) setErrCode:(NSString *)errCode
{
    JEDI_SYS_Try
    {
        [super setEntry:gs_opFather_errCode entry:errCode];
    }
    JEDI_SYS_EndTry
}

//------------------------------------------------------------------
-(NSString *) getErrMessage
{
    JEDI_SYS_Try
    {
        return [super getEntryString:gs_opFather_errMessage];
    }
    JEDI_SYS_EndTry
    
    return nil;
}

//------------------------------------------------------------------
-(void) setErrMessage:(NSString *)errMessage
{
    JEDI_SYS_Try
    {
        [super setEntry:gs_opFather_errMessage entry:errMessage];
    }
    JEDI_SYS_EndTry
}

//------------------------------------------------------------------
-(Boolean) isSuccessed
{
    JEDI_SYS_Try
    {
        return [super getEntryBoolean:gs_opFather_succeed];
    }
    JEDI_SYS_EndTry
    
    return false;
}

//------------------------------------------------------------------
-(void) setSuccess:(Boolean)successed
{
    JEDI_SYS_Try
    {
        [super setEntry:gs_opFather_succeed entry:[NSNumber numberWithBool:successed]];
    }
    JEDI_SYS_EndTry
}

//------------------------------------------------------------------
-(int) getUsedTime
{
    JEDI_SYS_Try
    {
        return [super getEntryInt:gs_opFather_usedTime];
    }
    JEDI_SYS_EndTry
    
    return 0;
}

//------------------------------------------------------------------
-(void) setUsedTime:(int)usedTime
{    
    JEDI_SYS_Try
    {
        [super setEntry:gs_opFather_usedTime entry:[NSNumber numberWithInt:usedTime]];
    }
    JEDI_SYS_EndTry
}

@end

File: WeChatService/EX02-CSTSv3-ClientAPI/src/jedi/ex01/CSTS3/comm/ipop/OPFather.java
package jedi.ex01.CSTS3.comm.ipop;

import allone.json.AbstractJsonData;

public class OPFather extends AbstractJsonData {

	public static final String jsonId = "opf";

	public static final String id = "-1";
	public static final String operateId = "-2";
	public static final String succeed = "-3";
	public static final String errCode = "-4";
	public static final String errMessage = "-5";
	public static final String usedTime = "-6";

	public OPFather() {
		super();
	}

	public OPFather(IPFather ip) {
		super();
		setEntry(OPFather.id, ip.getID());
		setEntry(OPFather.operateId, ip.getOperateId());
	}

	public String getID() {
		try {
			String data = getEntryString(OPFather.id);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public String getOperateId() {
		try {
			String data = getEntryString(OPFather.operateId);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public String getErrCode() {
		try {
			String data = getEntryString(OPFather.errCode);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setErrCode(String errCode) {
		setEntry(OPFather.errCode, errCode);
	}

	public String getErrMessage() {
		try {
			String data = getEntryString(OPFather.errMessage);
			return data;
		} catch (RuntimeException e) {
			return null;
		}
	}

	public void setErrMessage(String errMessage) {
		setEntry(OPFather.errMessage, errMessage);
	}

	public boolean isSucceed() {
		try {
			boolean data = getEntryBoolean(OPFather.succeed);
			return data;
		} catch (RuntimeException e) {
			return false;
		}
	}

	public void setSucceed(boolean succeed) {
		setEntry(OPFather.succeed, succeed);
	}

	public long getUsedTime() {
		try {
			long data = getEntryLong(OPFather.usedTime);
			return data;
		} catch (RuntimeException e) {
			return 0;
		}
	}

	public void setUsedTime(long usedTime) {
		setEntry(OPFather.usedTime, usedTime);
	}

}
